src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -105,14 +85,33 @@ private ToolConfigurationAndExecutablePath ObtainPackage()
                     configFile = new FilePath(_configFilePath);
                 }
 
-                return _toolPackageObtainer.ObtainAndReturnExecutablePath(
+                var obtainTransaction = _toolPackageObtainer.CreateObtainTransaction(
                     packageId: _packageId,
                     packageVersion: _packageVersion,
                     nugetconfig: configFile,
                     targetframework: _framework,
                     source: _source);
-            }
 
+                using (var t = new TransactionScope())
+                {
+                    Transaction.Current.EnlistVolatile(obtainTransaction, EnlistmentOptions.None);
```

Comment:
using (var t = new TransactionScope())
 Transaction.Current.EnlistVolatile(obtainTransaction, EnlistmentOptions.None);

are ceremonies, but it get things done. I think it is easier to read than hand roll.